src/lib/utils.h

```diff
@@ -56,6 +59,24 @@
         }                                                      \
     } while (/* CONSTCOND */ 0)
 
+/*
+ * @params
+ * array:       array of the structures to lookup
+ * id_field     name of the field to compare against
+ * ret_field    filed to return
+ * id           id to look for
+ * ret          return value
+ */
+#define LOOKUP_ARRAY_BY_ID(array, id_field, ret_field, id, ret) \
```

Comment:
The naming here is awkward to me. Something like `ARRAY_SEARCH_BY_INT` or similar would make more sense.

The way it currently is, it sounds like we're looking up an array somewhere, but that's not really the case.

Reply:
``ARRAY_SEARCH_BY_ID`` as it doesn't have to be an int